ignite/_utils.py

```diff
@@ -11,41 +11,24 @@ def _to_hours_mins_secs(time_taken):
     return hours, mins, secs
 
 
-def to_variable(batch, cuda=False, volatile=False):
-    if torch.is_tensor(batch):
-        if cuda:
-            batch = batch.cuda()
-        return Variable(batch, volatile=volatile)
-    elif isinstance(batch, string_classes):
-        return batch
-    elif isinstance(batch, collections.Mapping):
-        return {k: to_variable(sample, cuda=cuda, volatile=volatile) for k, sample in batch.items()}
-    elif isinstance(batch, collections.Sequence):
-        return [to_variable(sample, cuda=cuda, volatile=volatile) for sample in batch]
+def convert_tensor(input_, requires_grad=False, device=None):
+    if torch.is_tensor(input_):
+        if device:
+            input_ = input_.to(device=device)
+        if requires_grad != input_.requires_grad:
+            input_.requires_grad_(requires_grad)
+        return input_
+    elif isinstance(input_, string_classes):
+        return input_
+    elif isinstance(input_, collections.Mapping):
+        return {k: convert_tensor(sample, requires_grad=requires_grad, device=device) for k, sample in input_.items()}
+    elif isinstance(input_, collections.Sequence):
+        return [convert_tensor(sample, requires_grad=requires_grad, device=device) for sample in input_]
     else:
-        raise TypeError(("batch must contain tensors, numbers, dicts or lists; found {}"
-                         .format(type(batch[0]))))
-
-
-def to_tensor(variable, cpu=False):
-    if isinstance(variable, Variable):
-        t = variable.data
-        if cpu:
-            t = t.cpu()
-        return t
-    elif isinstance(variable, string_classes):
-        return variable
-    elif isinstance(variable, collections.Mapping):
-        return {k: to_tensor(sample, cpu=cpu) for k, sample in variable.items()}
-    elif isinstance(variable, collections.Sequence):
-        return [to_tensor(sample, cpu=cpu) for sample in variable]
-    else:
-        raise TypeError(("Argument variable must contain torch.autograd.Variable, numbers, dicts or lists; found {}"
-                         .format(type(variable))))
+        raise TypeError(("input must contain tensors, numbers, dicts or lists; found {}"
+                         .format(type(input_[0]))))
```

Comment:
Maybe  should be `input_[0]` -> `input_`, otherwise: 
```python
convert_tensor(12345)
```
fails with 
```
TypeError: 'int' object is not subscriptable
```